openvpn: T2146: remove old files

jjakob opened the pull request.

thomas-mangin commented on src/conf_mode/interfaces-openvpn.py:
```diff
@@ -899,6 +899,13 @@ def generate(openvpn):
     directory = os.path.dirname(get_config_name(interface))
 
+    # we can't know which clients were deleted, remove all client configs
+    if os.path.isdir(directory + '/ccd/' + openvpn['intf']):
```
This works but a better way would be to `os.path.join(directory,'ccd',openvpn['intf'])`

jjakob replied:
Fixed

c-po commented:
Looks good to me, too - is it still a WIP?

jjakob committed:
openvpn: T2146: delete old client configs

Previously old client configs for clients that were deleted from
the server stayed in the ccd directory, causing them to still be
used. As we can't know which clients were deleted, this deletes
all the client configs as they are recreated shortly later.

jjakob committed:
openvpn: T2146: remove user/pass auth file when not needed

jjakob commented:
Ready to merge.

jjakob commented:
@Elbandi Indeed, I didn't think of that at the time. It would require more code to list all the files in the directory, find which ones aren't configured clients, and delete just those. The script recreates all the ccd files from scratch soon after so it's not a problem just to delete them all. The server won't read them until being restarted anyway.
